### DISCUSSION TITLE
Question about using ISkSVGNode

### DISCUSSION THREAD

**Discussion by alie-samuel** [skia4delphi/skia4delphi#165] [Q&A] [answered]
Is there a way to modify the text value using  ISkSVGNode?

For example:
Original:
<'text id="test" x="87632" y="16662"  fill="black" font-weight="bold" font-size="10214px" font-family="Segoe UI FontID0">10</text'>
Modified
<'text id="test" x="87632" y="16662"  fill="black" font-weight="bold" font-size="10214px" font-family="Segoe UI FontID0">500</text'>

**Comment by viniciusfbb** [comment, marked as answer]
Hi @devsa707!

No. Skia renders and creates SVG very well but editing is very limited. Currently it only supports attribute editing.

However, for your specific text case, I should warn you that it might not be a good idea, as choosing the text string instead of the text vector will generate different renderings on different devices. Therefore, I prefer to leave the text vector in my SVG instead of the <text> tag.

Maybe your case can be solved by removing that text, and using Skia to create a new SVG, in which you would draw the original SVG without the <text> tag and then you would draw some text. These drawing commands Skia converts to SVG, so in the end you will have an original SVG without <text> tag + new text. And you still have the option to select if you want to export the vectorized texts.

See more in our SVG documentation: https://github.com/skia4delphi/skia4delphi/blob/main/Documents/SVG.md

**Reply by alie-samuel** [reply]
Hi @viniciusfbb thanks for the answer! 
I was trying to do a Chart for this specific case, i will try to draw a svg and align in the correclty position of the screen.


![image](https://user-images.githubusercontent.com/101226180/196042751-b43f6646-9d4d-44c8-9827-aa220b6c5d63.png)

**Reply by viniciusfbb** [reply]
Maybe in your case it's better to adopt a simpler solution, like an XML editor (which comes with Delphi), or even use some markup text: <text>%text1%</text>
and then you give a SvgString.Replace('%text1%', 'new text');

**Reply by alie-samuel** [reply]
it works well replacing the string, thank you!
